lib/matplotlib/axes/_axes.py

```diff
@@ -2441,10 +2495,8 @@ def bar(self, x, height, width=0.8, bottom=None, *, align="center",
         bar. See :doc:`/gallery/lines_bars_and_markers/bar_stacked`.
         """
         kwargs = cbook.normalize_kwargs(kwargs, mpatches.Patch)
-        color = kwargs.pop('color', None)
-        if color is None:
-            color = self._get_patches_for_fill.get_next_color()
-        edgecolor = kwargs.pop('edgecolor', None)
+        color, edgecolor = self._parse_bar_color_args(kwargs)
```

Comment:
```suggestion
        facecolor, edgecolor = self._parse_bar_color_args(kwargs)
```
Let's make this explicit and change the naming from here on - needs updates below.

Also note: It's used below in
```
color = itertools.chain(itertools.cycle(mcolors.to_rgba_array(color))
```
You can now leave out the `to_rgba_array()` there, because your helper already does the conversion.